Move plone.app.caching documentation to deployment/caching

ericof opened the pull request:
- [X] Move existing documentation from `plone.app.caching`
- [X] Convert documentation from RST to Markdown
- [X] Remove outdated references to kss
- [X] Add am introduction about HTTP Caching

ericof committed:
Move plone.app.caching documentation to deployment/caching

ericof committed:
Reference the term instead of linking to Varnish site

sneridagh commented on docs/deployment/caching/control-panel.md:
```diff
@@ -0,0 +1,63 @@
+
+Where you can control Plone's use of a caching proxy such or Varnish or a CDN.
+
+### `In-memory caache`
```
typo

sneridagh commented on docs/deployment/caching/composite-views.md:
```diff
@@ -0,0 +1,19 @@
+# Composite views
```
myst meta missing?

sneridagh reviewed (dismissed):
Minor typos, the rest LGTM

jensens commented on docs/deployment/caching/composite-views.md:
```diff
@@ -0,0 +1,19 @@
+
+However, most of the inline resources linked to from the composite view (css, javascript, images, etc.) can be cached very well in proxy so the overall speed of most composite views will always be better with a caching proxy in front despite the page itself not being cached.
+
+For relatively stable composite views or for those views for which you can tolerate some potential staleness, you might be tempted to try switching from *weak caching* to *moderate caching* with the `s-maxage` expiration value set to some tolerable value but first make sure you understand the issues regarding "split view" caching (see below).
```
rephrased a bit, the original is hard to understand and not 100% correct.

```suggestion
The term *composite view* refers to most of the page views that are visible on a Plone website, including views of content listings, content folders, many template views, and rarely also specific content items. 

These views can be difficult to manage because of the various changes that may affect the final composite view. As a result, composite views are typically challenging to remove reliably from caching proxies. 
To address this issue, the default caching profiles set headers that expire the cache immediately, which is known as *weak caching*.

However, the inline resources linked to from the composite view, such as CSS, JavaScript, and images, can typically be cached effectively in proxies. As a result, caching proxies can improve the overall speed of most composite views, even if the page itself is not cached.

For relatively stable composite views or views where some staleness is tolerable, it may be tempting to switch from *weak caching* to *terse caching* with the s-maxage expiration value set 60 seconds by default. 
```
- mention terse caching
- there is no split view caching, so removed this

jensens commented on docs/deployment/caching/composite-views.md:
```diff
@@ -0,0 +1,19 @@
+
+For relatively stable composite views or for those views for which you can tolerate some potential staleness, you might be tempted to try switching from *weak caching* to *moderate caching* with the `s-maxage` expiration value set to some tolerable value but first make sure you understand the issues regarding "split view" caching (see below).
+
+A way to speedup a site with lots of composite view is to use "Terse" caching.
+
+It caches a page for just a small time (be default 10s in browser and 60s in the caching proxy).
+
+Thus the content is almost up to date, but on high load visitors are getting pages served from the cache.
```
To increase the speed of a site that contains many composite views, a technique called "Terse" caching can be used. 
Terse caching involves caching a page for a brief period, typically 10 seconds in the browser and 60 seconds in the caching proxy. 
This ensures that the content is almost up to date, but during periods of high traffic, visitors may receive pages served from the cache.
This technique can actually speed up a site during periods of high traffic compared to low traffic since, during high traffic, the request can be served from the cache instead of having to go to the backend.

jensens replied:
```suggestion
To increase the speed of a site that contains many composite views, a technique called "Terse" caching can be used. 
Terse caching involves caching a page for a brief period, typically 10 seconds in the browser and 60 seconds in the caching proxy. 
This ensures that the content is almost up to date, but during periods of high traffic, visitors may receive pages served from the cache.
This technique can actually speed up a site during periods of high traffic compared to low traffic since, during high traffic, the request can be served from the cache instead of having to go to the backend.
```

jensens commented on docs/deployment/caching/enable.md:
````diff
@@ -0,0 +1,51 @@
+:alt: Caching Control Panel with caching disabled
+```
+
+To enable caching, just click the checkbox:
````
```suggestion
To enable caching, click the checkbox:
```

jensens commented on docs/deployment/caching/etags.md:
```diff
@@ -0,0 +1,75 @@
+
+# ETags
+
+ETags are used in to check whether pages need to be re-calculated or can be served from cache.
```
```suggestion
ETags are used in to determine whether pages need to be re-calculated or can be served from cache.
```

jensens commented on docs/deployment/caching/etags.md:
```diff
@@ -0,0 +1,75 @@
+# ETags
+
+ETags are used in to check whether pages need to be re-calculated or can be served from cache.
+An ETag is simply a string. Under `plone.app.caching`, it is a string of tokens separated by pipe characters.
+The tokens hold values such as a user id, the current skin name, or a counter indicating how many objects have been added to the site.
```
```suggestion
An ETag generated by `plone.app.caching' is a string of tokens separated by pipe characters that hold values like a user ID, the current skin name, or a counter indicating how many objects have been added to the site.
```

jensens commented on docs/deployment/caching/etags.md:
```diff
@@ -0,0 +1,75 @@
+ETags are used in to check whether pages need to be re-calculated or can be served from cache.
+An ETag is simply a string. Under `plone.app.caching`, it is a string of tokens separated by pipe characters.
+The tokens hold values such as a user id, the current skin name, or a counter indicating how many objects have been added to the site.
+The idea is that the browser sends a request with the ETag included in an `If-None-Match` header.
+Plone can then quickly calculate the current ETag for the requested resource.
+If the ETag is the same, Plone can reply with `304 NOT MODIFIED` response, telling the browser to use its cached copy.
+Otherwise, Plone renders the page and returns it as normal.
```
```suggestion
This technique is used to quickly calculate the current ETag for the requested resource. 
The browser sends a request with the ETag included in an `If-None-Match` header, and if the ETag is the same, Plone responds with a 304 NOT MODIFIED message, indicating that the browser should use its cached copy. 
If the ETag is different, Plone renders the page and returns it as normal.
```

jensens commented on docs/deployment/caching/etags.md:
```diff
@@ -0,0 +1,75 @@
+If the ETag is the same, Plone can reply with `304 NOT MODIFIED` response, telling the browser to use its cached copy.
+Otherwise, Plone renders the page and returns it as normal.
+
+Many caching operations use ETags. The tokens to include are typically listed in an `etags` tuple in the operation's options.
```
```suggestion
Many caching operations use ETags. 
The tokens to include are typically listed in an `etags` tuple in the operation's options.
```

jensens commented on docs/deployment/caching/etags.md:
```diff
@@ -0,0 +1,75 @@
+
+This should be a named adapter on the published object (typically a view, file resource or Zope page template object) and request, with a unique name.
+
+The name is used to look up the component. Thus, you can also override one of the tokens above for a particular type of context or request (e.g. via a browser layer), by registering a more specific adapter with the same name.
```
```suggestion
The name is used to look up the component. 
Thus, you can also override one of the tokens above for a particular type of context or request (e.g. via a browser layer), by registering a more specific adapter with the same name.
```

jensens commented on docs/deployment/caching/etags.md:
```diff
@@ -0,0 +1,75 @@
+from zope.interface import Interface
+
+@implementer(IETagValue)
+@adapter(Interface, Interface)
```
This generic adaption on Interface/Interface is a bad example.

ericof replied:
Agreed. I just copied from the original documentation

jensens commented on docs/deployment/caching/index.md:
```diff
@@ -0,0 +1,41 @@
+
+## Overview
+
+HTTP caching is a technique used to speed up the delivery of web content by storing previously requested resources (such as images, scripts, and stylesheets) in a cache. When a user requests a resource, the cache can serve the resource directly from its storage, rather than having to fetch it from the original source.
```
```suggestion
HTTP caching is a technique used to speed up the delivery of web content by storing previously requested resources (such as images, scripts, and stylesheets) in a cache. 
When a user requests a resource, the cache can serve the resource directly from its storage, rather than having to fetch it from the original source.
```

jensens commented on docs/deployment/caching/index.md:
```diff
@@ -0,0 +1,41 @@
+
+HTTP caching is a technique used to speed up the delivery of web content by storing previously requested resources (such as images, scripts, and stylesheets) in a cache. When a user requests a resource, the cache can serve the resource directly from its storage, rather than having to fetch it from the original source.
+
+It is possible to use a web accelerator, like {term}`Varnish` or {term}`Squid`, to implement HTTP caching in your own premises. The usual setup will have the web accelerator installed between Plone and the Internet, and when a request is made to the site the web accelerator will intercept the request and check to see if it has a cached copy of the requested resource. If a cached copy is found, th ...
```
Do we really want to mention Squid these days? 
Also here: split to one sentence per line
```suggestion
It is possible to use a web accelerator, like {term}`Varnish` to implement HTTP caching in your own premises. 
The usual setup will have the web accelerator installed between Plone and the Internet, and when a request is made to the site the web accelerator will intercept the request and check to see if it has a cached copy of the requested resource. 
If a cached copy is found, the accelerator will serve the cached copy directly to the client, otherwise it will make a request to the backend Plone server and then store a copy on the content in its local cache.
```

jensens commented on docs/deployment/caching/index.md:
```diff
@@ -0,0 +1,41 @@
+
+It is possible to use a web accelerator, like {term}`Varnish` or {term}`Squid`, to implement HTTP caching in your own premises. The usual setup will have the web accelerator installed between Plone and the Internet, and when a request is made to the site the web accelerator will intercept the request and check to see if it has a cached copy of the requested resource. If a cached copy is found, th ...
+
+A Content Delivery Network (CDN) can also be used to implement the HTTP caching and speed up the delivery of web content. A CDN is a network of servers located in multiple geographic regions that work together to deliver web content to users more quickly and efficiently. When a user requests a resource from a website that is using a CDN, the request is routed to the nearest server in the CDN, rat ...
```
```suggestion
HTTP caching and the speed of delivering web content can be improved using a Content Delivery Network (CDN).
A CDN is a network of servers located in various geographic regions that work together to deliver web content to users quickly and efficiently. 
When a user requests a resource from a website that uses a CDN, the request is directed to the closest server in the CDN, rather than having to travel all the way to the website's origin server.
```

jensens commented on docs/deployment/caching/index.md:
```diff
@@ -0,0 +1,41 @@
+
+## Plone support
+
+Plone, since version 4.1, ships with a powerful and extensible `HTTP` and `In-Memory` cache support, implemented by the package `plone.app.caching`.
```
I do not think we need a history lesson here ;)
```suggestion
Plone ships with a powerful and extensible `HTTP` and `In-Memory` cache support, implemented by the package `plone.app.caching`.
```

jensens commented on docs/deployment/caching/profiles.md:
```diff
@@ -0,0 +1,116 @@
+
+## Default caching profiles
+
+`plone.app.caching` includes three default caching profiles.
+
+Two of these  profiles encapsulate the cache settings that are known to work well with a typical default Plone installation:
```
```suggestion
`plone.app.caching` includes two default caching profiles.
These  profiles encapsulate the cache settings that are known to work well with a typical default Plone installation:
```

jensens commented on docs/deployment/caching/profiles.md:
```diff
@@ -0,0 +1,116 @@
+
+### With caching proxy
+
+Settings useful for setups with a caching proxy such as Squid or Varnish.
```
```suggestion
Settings useful for setups with a caching proxy such Varnish or a CDN.
```

jensens commented on docs/deployment/caching/profiles.md:
```diff
@@ -0,0 +1,116 @@
+### With caching proxy
+
+Settings useful for setups with a caching proxy such as Squid or Varnish.
+The only difference from the "without caching proxy" profile are some settings to enable proxy caching of files/images in content space and content feeds.
```
I am pretty sure this is not true anymore.

ericof replied:
I just checked the profiles, and it is it. Feed and Files use `moderateCaching` with `smaxage`:

## With Proxy
```xml
  <!-- plone.content.file -->
  <record name="plone.app.caching.moderateCaching.plone.content.file.smaxage">
    <field ref="plone.app.caching.moderateCaching.smaxage" />
    <value>86400</value>
  </record>
  <record name="plone.app.caching.moderateCaching.plone.content.file.lastModified">
    <field ref="plone.app.caching.moderateCaching.lastModified" />
    <value>True</value>
  </record>
```

## Without Proxy
```xml
  <record name="plone.app.caching.weakCaching.plone.content.file.lastModified">
    <field ref="plone.app.caching.weakCaching.lastModified" />
    <value>True</value>
  </record>
```

jensens commented on docs/deployment/caching/proxies.md:
```diff
@@ -0,0 +1,108 @@
+
+Since resources are cached in the proxy based on their URL, you can "invalidate" the cached copy by changing an item's URL when it is updated.
+
+This is the approach taken by Plone's resource management: In production mode, the links that are inserted into Plone's content pages for resource managed by Plone  contain a hash-based token, which changes when the main bundle file changes.
```
```suggestion
This is the approach taken by Plone's resource management: Links that are inserted into Plone's content pages for resources managed by Plone  contain a hash-based token, which changes when the main bundle file changes.
```

jensens commented on docs/deployment/caching/proxies.md:
```diff
@@ -0,0 +1,108 @@
+
+Given the proxy has cached a resource at `/logo.jpg`, and that object is modified.
+
+Then a PURGE request could be sent to the proxy (originating from Zope, not the client) with the same path to force the proxy to fetch a new version the next time the item is requested.
```
```suggestion
Then a PURGE request could be sent to the proxy (originating from the backend, not the browser) with the same path to force the proxy to fetch a new version the next time the item is requested.
```

jensens commented on docs/deployment/caching/proxies.md:
```diff
@@ -0,0 +1,108 @@
+
+## Purging a caching proxy
+
+Synchronous and asynchronous purging is enabled via [plone.cachepurging](https://pypi.org/project/plone.cachepurging).
```
```suggestion
Synchronous and asynchronous purging is enabled via [plone.cachepurging](https://pypi.org/project/plone.cachepurging), which is installed as a dependency of `plone.app.caching`.
```

jensens commented on docs/glossary.md:
```diff
@@ -635,4 +635,10 @@ traversal
+Varnish
+    [Varnish](https://varnish-cache.org) is a popular open-source web accelerator that is used to implement HTTP caching.
+
+Squid
```
I would kick out Squid everywhere.

jensens reviewed (changes requested):
Good to have the caching docs here now. Do not forget to remove them from plone.app.caching and set a link to the Plone docs. 

Overall there are many newlines between the sentences. This results in many paragraphs, is this by intend? 

Further, see my comments.

ericof committed:
Add myst meta information to composite views page

ericof committed:
Fix typo

ericof committed:
Apply suggestions from jensens code review

Co-authored-by: Jens W. Klein <jk@kleinundpartner.at>

ericof committed:
Remove references to Squid

jensens commented on docs/deployment/caching/etags.md:
```diff
@@ -0,0 +1,75 @@
+# ETags
+
+ETags are used in to determine whether pages need to be re-calculated or can be served from cache.
+An ETag generated by `plone.app.caching' is a string of tokens separated by pipe characters that hold values like a user ID, the current skin name, or a counter indicating how many objects have been added to the site.
```
```suggestion
An ETag generated by `plone.app.caching` is a string of tokens separated by pipe characters that hold values like a user ID, the current skin name, or a counter indicating how many objects have been added to the site.
```

jensens committed:
Update docs/deployment/caching/etags.md

jensens commented on docs/deployment/caching/etags.md:
```diff
@@ -0,0 +1,75 @@
+# ETags
+
+ETags are used in to determine whether pages need to be re-calculated or can be served from cache.
+An ETag generated by `plone.app.caching` is a string of tokens separated by pipe characters that hold values like a user ID, the current skin name, or a counter indicating how many objects have been added to the site.
```
just found: "the current skin name" - is this nowadays a good example?

ericof replied:
This is still in `plone.app.caching`

ericof committed:
Merge branch '6-dev' into issue-1448-caching

ericof committed:
Merge branch '6-dev' into issue-1448-caching

ericof commented:
@stevepiercy Any idea why the tests are failing?

ksuess commented:
> @stevepiercy Any idea why the tests are failing?

Tests are green again after re-running.

stevepiercy commented:
The PR from @fredvd that updated GitHub Actions broke the build after I merged it. I did not watch the build fail, and ignored the warning without checking its details, assuming it was ogp.me still failing under linkcheckbroken. I notified @fredvd at 5:10 AM Pacific, and he quickly responded by pushing [fixes to `6-dev`](https://github.com/plone/documentation/commits/6-dev). Everything should be good now. Sorry for the inconvenience.

Lesson learned: merge a PR only if you have time to watch it build successfully, and can fix it yourself if it fails.

fredvd commented:
The weird part about why it failed was that yarn got upgraded to v3  during/after the PR.  so the $(yarn cache dir) no longer worked.  

Work on the workflows willstabose soonish, unfortunately small changes here can have big impact.

stevepiercy commented:
Please do not merge. I am doing a review, and will push fixes soon.

stevepiercy committed:
No hyphen needed

stevepiercy committed:
Clean up display of indices in navigation

stevepiercy committed:
Clean up of caching/index.md
- We cache content in Plone, not Plone itself.
- Headings for overview or introduction are unnecessary, and leaving a heading without content looks weird. Instead just smoosh an introduction under the page title.
- Shorten introduction of caching.
- Add Glossary term of CDN.
- Add link to plone.app.caching repo.
- Grammar fixes.

stevepiercy committed:
Clean up of caching/enable.md
- Use shorter sentences.
- Use MyST syntax for GUI elements.
- Use Sphinx Design card feature to display screenshots with a caption and target to open a full-size image.
- Redo screenshots for a 740-pixel width viewport

stevepiercy committed:
Clean up of caching/control-panel.md
- Use MyST syntax for GUI elements.
- Use Sphinx Design card feature to display screenshots with a caption and target to open a full-size image.
- Fix grammar.

stevepiercy committed:
Clean up of caching/profiles.md
- Use MyST syntax for GUI elements.
- Fix grammar.

stevepiercy committed:
Remove replaced images

stevepiercy committed:
Add cross-referenced links to rulesets

stevepiercy committed:
Ignore the `/_static/` directory for linkcheckbroken

stevepiercy commented on docs/deployment/caching/rulesets-and-caching-operations.md:
```diff
@@ -0,0 +1,300 @@
+
+Stable resources like the css, and javascript files registered with the Resource Registries.  These are resources which can be cached 'forever'.  Normally this means that if the object does change, its URL changes too.
+
+## Default cache operations
```
I don't see any of these default cache operations on the [Plone Classic Demo site](https://classic.demo.plone.org/@@caching-controlpanel#autotoc-item-autotoc-3). Where are they?

jensens replied:
The demo site has caching installed, but not set up. I now set it up by importing the `Without caching proxy` profile and then enabled caching - then all is there.
(this will be auto-reset soon).

stevepiercy replied:
OK, I see them now. This process needs to be documented. Newbies like me have no clue.

Here is a very rough outline, but to be honest, I have no idea what I'm doing. Maybe someone with more familiarity can fill in the details?

First, enable caching, from `caching/enable.md`.

Then:

https://classic.demo.plone.org/@@caching-controlpanel-import

Advanced > Import caching profiles > select Without caching proxy

Then:

https://classic.demo.plone.org/@@caching-controlpanel#autotoc-item-autotoc-4

Advanced > Caching settings > Detailed settings

Then:

- [View/edit](https://classic.demo.plone.org/@@caching-controlpanel/edit-operation-global/plone.app.caching.weakCaching) global parameters
- [View/edit/clear](https://classic.demo.plone.org/@@caching-controlpanel/edit-operation-ruleset/plone.app.caching.weakCaching/plone.content.feed) per-ruleset parameters

ericof replied:
@stevepiercy There is a message on top of the page: 
`First time here? We recommend that you get started by importing a preconfigured set of caching rules`

stevepiercy replied:
OK, but how does a newbie do that? This single sentence does not tell a newbie how to do it, which is why I suggested the above outline.

jensens replied:
Detailed instructions are missing in `enable.md`, which we could refer to here.

stevepiercy replied:
I'm fine with wherever it belongs. I need help fleshing out the details of the suggested outline.

jensens replied:
After (or better before) enable, use the controlpanle to import a profile:
![screenshot_2023-02-25_23:24:02_selection](https://user-images.githubusercontent.com/157140/221382310-8ad49446-b669-467a-902a-af1b794ed06b.png)

jensens replied:
forgot to say: no need to do more, no advanced/ZMI, nothing else, two steps. I usually import one of the profiles, depending on the setup and then click enable.

stevepiercy replied:
Resolved in [87dcb50](https://github.com/plone/documentation/pull/1449/commits/87dcb5062e646f50d3486e371d0398bff6529133). Please review. Preview: https://63fb44f1bf95c20008d66c9d--6-docs-plone-org.netlify.app/deployment/caching/rulesets-and-caching-operations.html, then click through to new and existing sections.

jensens replied:
good, except (see review below) the mention of Squid sneaked in again.

stevepiercy reviewed (commented):
A question for now, while I continue my review.

stevepiercy committed:
Tidy up rulesets-and-caching-operations.md

stevepiercy commented on docs/deployment/caching/rulesets-and-caching-operations.md:
```diff
@@ -0,0 +1,320 @@
+    In this case, you can either declare the ruleset on the type as shown above (in ZCML), or map the type name in the registry, using the GUI or GenericSetup.
+    The former is more robust and certainly more natural if you are declaring other, more conventional rulesets in {term}`ZCML` already.
+-   You want to set the ruleset for a number of content types.
+    In fact, `plone.app.caching` already does this for you: The Archetypes base classes `BaseContent` and `BaseFolder` are assigned the rulesets `plone.content.itemView` and `plone.content.folderview`, respectively.
```
Remove Archetypes and merge this sentence with the next one.

stevepiercy replied:
This is the last blocker. I'm not sure of wording, but here is what I would suggest.

```
    In fact, `plone.app.caching` already does this for you.
    Dexterity `IDexterityItem` and `IDexterityContainer` interfaces are assigned the rulesets `plone.content.itemView` and `plone.content.folderview`, respectively.
```

@ericof @jensens Please let me know if this is in fact correct, and we can commit and merge this PR. Thank you!

jensens replied:
Yes, see https://github.com/plone/plone.app.caching/blob/master/plone/app/caching/caching.zcml#L106-L114

stevepiercy replied:
Done in 0dab350b6ad27f3f4a307af95f4b46774e18eb06

stevepiercy committed:
Revert "Clean up of caching/control-panel.md"

This reverts commit b5c2706b40549bb641d4b5e25e9bf0208e9f1806.

stevepiercy committed:
Restore control-panel.md changes without git submodules

stevepiercy committed:
Remove unused image

stevepiercy committed:
Tidy up proxies.md

stevepiercy committed:
Tidy up code samples

stevepiercy committed:
Tidy up ram-cache.md

stevepiercy committed:
Tidy up etags.md

stevepiercy committed:
Tidy up composite-views.md

stevepiercy committed:
Tidy up restapi.md

stevepiercy commented:
My review is now complete. Please respond to my question and my suggestion. Thank you!

jensens committed:
Merge branch '6-dev' into issue-1448-caching

stevepiercy committed:
Merge branch '6-dev' into issue-1448-caching

stevepiercy reviewed (dismissed):
Please remove:

- `docs/_static/caching/ControlPanel-03.png`
- submodules

stevepiercy committed:
Code block lexer of http does not work, use text instead

stevepiercy committed:
Add new section to import caching profiles and reference from rulesets-and-caching-operations.md
Refs: https://github.com/plone/documentation/pull/1449#discussion_r1117986845

jensens commented on docs/deployment/caching/enable.md:
```diff
@@ -0,0 +1,142 @@
+Select options that are appropriate for your caching environment.
+
+{guilabel}`With caching proxy`
+:   Settings useful for setups with a caching proxy such as Squid or Varnish
```
```suggestion
:   Settings useful for setups with a caching proxy such as Varnish or a CDN
```

jensens commented on docs/deployment/caching/enable.md:
```diff
@@ -0,0 +1,142 @@
+:   Settings useful for setups with a caching proxy such as Squid or Varnish
+
+{guilabel}`Without caching proxy`
+:   Settings useful for setups without a caching proxy such as Squid or Varnish
```
```suggestion
:   Settings useful for setups without a caching proxy such as Varnish or a CDN
```

jensens commented on docs/deployment/caching/profiles.md:
```diff
@@ -0,0 +1,111 @@
+<genericsetup:registerProfile
+    name="with-caching-proxy"
+    title="With caching proxy"
+    description="Settings useful for setups with a caching proxy such as Squid or Varnish"
```
```suggestion
    description="Settings useful for setups with a caching proxy such as Varnish or a CDN"
```

jensens reviewed (changes requested):
Do not mention outdated Squid.

jensens committed:
Merge branch '6-dev' into issue-1448-caching

stevepiercy committed:
Update docs/deployment/caching/enable.md

Co-authored-by: Jens W. Klein <jk@kleinundpartner.at>

stevepiercy committed:
Update docs/deployment/caching/enable.md

Co-authored-by: Jens W. Klein <jk@kleinundpartner.at>

stevepiercy committed:
Update docs/deployment/caching/profiles.md

Co-authored-by: Jens W. Klein <jk@kleinundpartner.at>

stevepiercy commented:
@jensens @ericof please see the final blocker to merging this PR:

https://github.com/plone/documentation/pull/1449#discussion_r1118070026

stevepiercy committed:
Remove Archetypes mention

stevepiercy commented:
One more final review @jensens, if you please. Thank you!
